lib/libzfs/libzfs_sendrecv.c

```diff
@@ -610,6 +610,17 @@ send_iterate_fs(zfs_handle_t *zhp, void *arg)
 	fnvlist_free(sd->snapprops);
 	fnvlist_free(sd->snapholds);
 
+	/* Do not allow the size of the properties list to exceed the limit */
+	if ((fnvlist_size(nvfs) + fnvlist_size(sd->fss)) >
+	    (SPA_MAXBLOCKSIZE * 4)) {
+		(void) fprintf(stderr, dgettext(TEXT_DOMAIN,
+		    "warning: cannot send %s@%s: the size of the list of "
+		    "snapshots and properties is too large to be received "
+		    "successfully. Select a smaller number of snapshots to "
+		    "send.\n"), zhp->zfs_name, sd->tosnap);
```

Comment:
I'd suggest adding a `\n` to this message to make sure it wraps reasonably for 80 character terminals.

Reply:
This is maybe difficult because the size of the dataset and snapshot name are dynamic. Should we just put a line break at the sentence stop?

Reply:
That sounds reasonable.